custom/enikshay/management/commands/import_voucher_confirmations.py

```diff
@@ -0,0 +1,234 @@
+            ] + [
+                api_payload[prop] for prop in self.voucher_api_properties
+            ]
+
+        rows = map(make_row, unmodified_vouchers)
+        self.write_csv('updates', headers, rows)
+
+    def update_vouchers(self, voucher_updates):
+        print "updating voucher cases"
+        for chunk in chunked(with_progress_bar(voucher_updates), 100):
+            updates = [
+                (update.case_id, update.properties, False)
+                for update in chunk
+            ]
+            if self.commit:
+                bulk_update_cases(self.domain, updates)
+
+    def reconcile_repeat_records(self, voucher_updates):
+        """
+        Mark updated records as "succeeded", all others as "cancelled"
+        Delete duplicate records if any exist
+        """
+        print "Reconciling repeat records"
+        chemist_voucher_repeater_id = 'be435d3f407bfb1016cc89ebbf8146b1'
+        lab_voucher_repeater_id = 'be435d3f407bfb1016cc89ebbfc42a47'
+
+        already_seen = set()
+        updates_by_voucher_id = {update.id: update for update in voucher_updates}
+
+        headers = ['record_id', 'voucher_id', 'status']
+        rows = []
+
+        get_db = (lambda: IterDB(RepeatRecord.get_db())) if self.commit else MagicMock
+        with get_db() as iter_db:
+            for repeater_id in [chemist_voucher_repeater_id, lab_voucher_repeater_id]:
+                print "repeater {}".format(repeater_id)
+                records = iter_repeat_records_by_domain(self.domain, repeater_id=repeater_id)
+                record_count = get_repeat_record_count(self.domain, repeater_id=repeater_id)
+                for record in with_progress_bar(records, record_count):
+                    if record.payload_id in already_seen:
+                        status = "deleted"
+                        iter_db.delete(record)
+                    elif record.payload_id in updates_by_voucher_id:
+                        # add successful attempt
+                        status = "succeeded"
+                        attempt = RepeatRecordAttempt(
+                            cancelled=False,
+                            datetime=datetime.datetime.utcnow(),
+                            failure_reason=None,
+                            success_response="Paid offline via import_voucher_confirmations",
+                            next_check=None,
+                            succeeded=True,
+                        )
+                        record.add_attempt(attempt)
+                        iter_db.save(record)
+                    else:
+                        # mark record as canceled
+                        status = "cancelled"
+                        record.cancelled = True
+                        record.succeeded = False
+                        record.failure_reason = ''
+                        record.overall_tries = 0
+                        record.next_check = None
```

Comment:
How come you don't add a `RepeatRecordAttempt` here? I'd recommend always doing that as the mechanism for changing the top level vars.